### PR TITLE
Migrations config, run on startup, 12 factor setup

### PR DESCRIPTION
**Pull request by dbelwood**

### CODE REVIEW

File: migrations/sqls/20170908045555-add-triggers-down.sql [hunk 1]
```diff
@@ -0,0 +1,6 @@
+DROP TRIGGER changes_log ON offer_c;
```
Comment (jmlow) [inline migrations/sqls/20170908045555-add-triggers-down.sql]: Should this be 'changes_offer_log' for clarity?

File: migrations/sqls/20170908045555-add-triggers-up.sql [hunk 2]
```diff
@@ -0,0 +1,54 @@
+CREATE TABLE Offer_c(
```
Comment (jmlow) [inline migrations/sqls/20170908045555-add-triggers-up.sql]: Schemas and namespaces are missing for tables/triggers. TA78 will handle Salesforce schema, and TA77 needs to be merged in by Suresh.

File: src/index.ts [hunk 3]
```diff
@@ -0,0 +1,20 @@
+export enum SObjectTable {
+    OFFERS = "offer__c",
```
Comment (jmlow) [inline src/index.ts]: Add Product table as it's currently included in the migration.

File: migrations/sqls/20170913195016-add-tables-up.sql [hunk 4]
```diff
@@ -0,0 +1,18 @@
+CREATE SCHEMA IF NOT EXISTS changes;
+
+CREATE TABLE IF NOT EXISTS changes.Offer_c(
```
Comment (dbelwood) [inline migrations/sqls/20170913195016-add-tables-up.sql]: Please remove this table migration

File: migrations/sqls/20170913195016-add-tables-up.sql [hunk 5]
```diff
@@ -0,0 +1,18 @@
...
+	offer_id integer,
+    offer_name text
+);
+
+CREATE TABLE IF NOT EXISTS changes.products_c(
```
Comment (dbelwood) [inline migrations/sqls/20170913195016-add-tables-up.sql]: Please remove this table migration

File: migrations/sqls/20170913195536-add-triggers-up.sql [hunk 6]
```diff
@@ -0,0 +1,32 @@
...
+
+CREATE OR REPLACE FUNCTION changes.notify_trigger() RETURNS trigger AS $$
+DECLARE
+BEGIN
+  PERFORM pg_notify('watchers', '' || NEW.* || '' );
+  RETURN NEW;
+END;
+$$ LANGUAGE plpgsql;
+
+CREATE TRIGGER changes_log BEFORE INSERT OR UPDATE OR DELETE  ON changes.offer_c
```
Comment (dbelwood) [inline migrations/sqls/20170913195536-add-triggers-up.sql]: Can we move these triggers to a PR after the /dev migration from @mrockford

Comment (skandru) [reply migrations/sqls/20170913195536-add-triggers-up.sql]: Sure, @mrockford changes or schema creation that one I already added to tables_script . Is there any other changes we have to merge

File: migrations/sqls/20170913195016-add-tables-up.sql (hunk 5 above)
Comment (skandru) [reply migrations/sqls/20170913195016-add-tables-up.sql]: Is there any reason to remove or do we have other task for tables scripts?

File: migrations/sqls/20170913195016-add-tables-up.sql (hunk 4 above)
Comment (skandru) [reply migrations/sqls/20170913195016-add-tables-up.sql]: Removed tables migration.

File: migrations/sqls/20170913195536-add-triggers-up.sql [hunk 7]
```diff
@@ -0,0 +1,32 @@
...
+FOR EACH ROW EXECUTE PROCEDURE changes.changes_log();
+
+CREATE TRIGGER changes_product_log BEFORE INSERT OR UPDATE OR DELETE ON changes.products_c
```
Comment (dbelwood) [inline migrations/sqls/20170913195536-add-triggers-up.sql]: @skandru please remove these trigger definitions

File: migrations/sqls/20170913195536-add-triggers-down.sql [hunk 8]
```diff
@@ -0,0 +1,5 @@
+DROP TRIGGER changes_log ON changes.offer_c;
+
+DROP TRIGGER changes_product_log ON changes.products_c;
```
Comment (dbelwood) [inline migrations/sqls/20170913195536-add-triggers-down.sql]: Once you remove the trigger defns below, you can remove these DROP commands as well

File: migrations/sqls/20170913195536-add-triggers-up.sql (hunk 7 above)
Comment (skandru) [reply migrations/sqls/20170913195536-add-triggers-up.sql]: Sure Dan

File: migrations/sqls/20170913195536-add-triggers-down.sql (hunk 8 above)
Comment (skandru) [reply migrations/sqls/20170913195536-add-triggers-down.sql]: Removed

File: migrations/sqls/20170913195016-add-tables-up.sql [hunk 9]
```diff
@@ -0,0 +1,2 @@
+CREATE SCHEMA IF NOT EXISTS changes;
+
```
Comment (dbelwood) [inline migrations/sqls/20170913195016-add-tables-up.sql]: @mrockford  add the change table definition here as well

Comment (jmlow) [reply migrations/sqls/20170913195016-add-tables-up.sql]: @mrockford the name we decided on was changes_log

File: migrations/sqls/20170913195016-add-tables-down.sql [hunk 10]
```diff
@@ -0,0 +1,2 @@
+DROP SCHEMA changes CASCADE;
+
```
Comment (jmlow) [inline migrations/sqls/20170913195016-add-tables-down.sql]: @mrockford need to include a DROP Table for changes_log, to follow from the previous comments

Comment (skandru) [reply migrations/sqls/20170913195016-add-tables-down.sql]: @jmlow cascade will drop all tables inside schema and drop the schema. I think not required to mentioned specifically drop tables

Comment (jmlow) [reply migrations/sqls/20170913195016-add-tables-down.sql]: @skandru Good call. Totally spaced on that. Auto-pilot...

**Pull request merged**